source/common/network/udp_listener_impl.cc

```diff
@@ -31,31 +31,26 @@ namespace Network {
 UdpListenerImpl::UdpListenerImpl(Event::DispatcherImpl& dispatcher, SocketSharedPtr socket,
                                  UdpListenerCallbacks& cb, TimeSource& time_source)
     : BaseListenerImpl(dispatcher, std::move(socket)), cb_(cb), time_source_(time_source) {
-  file_event_ = socket_->ioHandle().createFileEvent(
+  socket_->ioHandle().createFileEvent(
       dispatcher, [this](uint32_t events) -> void { onSocketEvent(events); },
       Event::PlatformDefaultTriggerType, Event::FileReadyType::Read | Event::FileReadyType::Write);
 
-  ASSERT(file_event_);
-
   if (!Network::Socket::applyOptions(socket_->options(), *socket_,
                                      envoy::config::core::v3::SocketOption::STATE_BOUND)) {
     throw CreateListenerException(fmt::format("cannot set post-bound socket option on socket: {}",
                                               socket_->localAddress()->asString()));
   }
 }
 
-UdpListenerImpl::~UdpListenerImpl() {
-  disableEvent();
-  file_event_.reset();
```

Comment:
I am not sure here if we should expose this method in `IoHandle`. This will be useful in the following scenario:

- Single `IoHandle` instance
- Consumer 1, creates a file event -> does some work -> disposes it
- Consumer 2, creates a file event -> does some work -> disposes it

This scenario is not allowed with the current implementation even though it does not violate the rule of at most 1 file_event per fd

Reply:
Turns out we need to support it, otherwise a bunch of things crash.